ndc-sqlgen/src/main/kotlin/io/hasura/ndc/sqlgen/BaseGenerator.kt

```diff
@@ -125,16 +121,23 @@ sealed interface BaseGenerator {
                         splitCollectionName(getCollectionForCompCol(e.column, request)) + e.column.name
                     )
                 )
-                val comparisonValue = when (val v = e.value) {
+
+                when (val v = e.value) {
+                    is ComparisonValue.ScalarComp -> {
+                        if (e.operator == ApplyBinaryComparisonOperator.IN) {
+                            handleScalarComp(column, v)
+                        } else {
+                            buildComparison(column, e.operator, DSL.inline(v.value))
+                        }
+                    }
+                    is ComparisonValue.VariableComp -> {
+                        column.`in`(DSL.field(DSL.name(listOf("vars", v.name))))
```

Comment:
I'm not clear on the use of `in` here or in the `ColumnComp` context. Can you explain?